sdk/java/src/main/java/net/consensys/gpact/common/BlockchainConfig.java

```diff
@@ -19,23 +19,48 @@ public class BlockchainConfig {
   // Blockchain identifier.
   public BlockchainId bcId;
   // URI of the blockchain node to be used for connecting to the blockchain.
-  public String uri;
+  public String blockchainNodeRpcUri;
+  // Web socket URI of blockchain node to be used for connecting to the blockchain.
+  public String blockchainNodeWsUri;
+
   // Strategy to be used for setting gas prices.
   public DynamicGasProvider.Strategy gasPriceStrategy;
   // How often to poll the blockchain for a response. Typically set to the block period.
   public int period;
 
+  public String msgStoreUrlFromDispatcher;
+  public String msgStoreUrlFromUser;
+
+  public String dispatcherUri;
+
+  public String observerUri;
```

Comment:
it isn't immediately clear to me why the SDK needs to know about the observer and the dispatcher of one or more relayer? I would have expected these details to largely be transparent to the SDK? Am I missing something?

Reply:
The SDK needs this information so it can configure the Relayer / Attestor. The philosophy that I have taken is that customers will need to configure the nodes. They will be accustomed to the SDK, so it makes sense for it to be tool to use to configure it. Additionally, it allows the Relayer / Attestor to be set-up for testing.

The BlockchainConfig is passed into the Messaging Layer interfaces. I think separate to this PR, we need to think through how we want to configure the Relayer / Attestor, and what configurations we want to allow for (or allow for in a straight forward manner).

Reply:
ok. my 2cents here is that users writing GPACT applications will largely be distinct from those operating and validating a bridge. Even if there might exist overlap in certain scenarios (enterprise use-cases..), these feel like separate concerns to me. I think the risk we have with this approach is that it might complicate the SDK and tie knowledge of Relayer architecture (and deployment model) into it, which might limit how both these components can evolve. I believe having a separate toolset for deploying and configuring Relayers and keeping the SDKs focused on the core requirements for building applications might give us more simplicity and flexibility.  

we can defer this discussion/decision to a later point if you'd prefer.